index.html

```diff
@@ -456,105 +464,107 @@ <h2>
-              <var>p</var> with a <code>"<a>NotAllowedError</a>"</code>
-              <a>DOMException</a>, and abort these steps.
-            </li>
-          </ol>
-        </li>
-        <li>
-          Fulfill <var>p</var>.
-        </li>
-      </ol>
+      <section>
+        <h3><dfn>request()</dfn> method</h3>
+        <p data-tests="wakelock-api.https.html">
+          The <a>request()</a> method, when invoked, MUST run the
+           following steps:
+        </p>
+        <ol class="algorithm">
+          <li>
+            Let |promise| be a new <a>Promise</a> object.
+          </li>
+          <li>
+            If <a>obtain permission</a> steps for <code>this</code>'s
+            |type| return <code>false</code>,
+            then reject |promise| with a "<a>NotAllowedError</a>"
+            <a>DOMException</a>, and abort these steps.
+          </li>
+          <li>
+            Let |options| be the first argument.
+          </li>
+          <li>
+            Let |signal| be the |options|’ dictionary member
+            of the same name if present, or <code>null</code> otherwise.
+          </li>
+          <li>
+            Let |record| be the <a>platform wake lock</a>'s
+            <a>state record</a> associated with
+            <code>this</code>'s |type|.
+          </li>
+          <li>
+            If |signal|’s <a>aborted flag</a> is set, then
+            reject |promise| with an "<a>AbortError</a>"
+            <a>DOMException</a>, and abort these steps.
+          </li>
+          <li>
+            If |signal| is not <code>null</code>, then
+            <a>add the following abort steps</a> to |signal|:
+            <ol>
+              <li>
+                Decrease |record|.<a>[[\RequestCounter]]</a> by one.
+              </li>
+              <li>
+                If |record|.<a>[[\RequestCounter]]</a> is 0, then
+                run the following steps <a>in parallel</a>:
+                <ol>
+                  <li>
+                    Run <a>release a wake lock</a> on <code>this</code>.
+                  </li>
+                </ol>
+              </li>
+              <li>
+                Reject |promise| with an "<a>AbortError</a>"
+                <a>DOMException</a>, and abort these steps.
+              </li>
+            </ol>
+          </li>
+          <li>
+            If the <a>current global object</a> is not a <a>DedicatedWorkerGlobalScope</a>
+            object, run the following steps:
+            <ol>
+              <li>
+                Let |document| be the <a>responsible document</a> of the
+                <a>current settings object</a>.
+              </li>
+              <li>
+                If |document| is not <a>fully active</a>,
+                reject |promise| with a <code>"<a>NotAllowedError</a>"</code>
+                <a>DOMException</a>, and abort these steps.
+              </li>
+              <li>
+                If <code>this</code>'s |type| is
```

Comment:
```suggestion
                If <a>context object</a>'s |type| is
```

Reply:
Interesting... so always do this for `this`?

Reply:
Yes please :) See also in payment request spec. ... but do:


```
Let |lock| be the <a>context object</a>.
```

Otherwise, it's super super confusing to know what "this" is.